### PR TITLE
Added branches lesson

### LINKED ISSUE #1: Add branches lesson

### PR DESCRIPTION
**Pull request by sstevens2**
Added a draft of the branches lesson.  Also shifted all the other lesson numbers to make room for the branches after `.gitignore` and before github.

Has no challenges and could still use some work.

Closes #1

### CODE REVIEW

**Comment by pshriwise** [comment]
There appears to be some overlap here between the `Branches and Pull Requests` lesson and this new lesson. This was probably expected? We can address it in a future PR which separates that conent.

File: _episodes/07-branches.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
+---
+title: Branches
+teaching: 20
+exercises: 0
+questions:
+- "What are branches?"
+- "How can I work in parallel using branches?"
+objectives:
+- "Explain why branches can be useful."
```
Comment (pshriwise) [inline _episodes/07-branches.md]: Could this objective be something like "Understand why branches are useful for working on separate tasks in the same repository."?

File: _episodes/07-branches.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+keypoints:
+- "Branches can be useful for developing while keeping the main line static."
+---
+
+So far we've always been working in a straight timeline.
+However, there are times when we might want to keep
+our main work safe from experimental changes we are working on.
+To do this we can use branches to work in parallel from our main branch.
+
+We didn't see it before the the first branch made is called `master`.
```
Comment (pshriwise) [inline _episodes/07-branches.md]: Two the's here. One is supposed to be a "but"?

File: _episodes/07-branches.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+~~~
+* master
+~~~
+{: .output}
+
+
+The '*' indicates which branch we are currently on.
+
+In this lesson, Dracula is trying to run an analysis
```
Comment (pshriwise) [inline _episodes/07-branches.md]: I like this as an example concept!! Ties in Python nicely.

File: _episodes/07-branches.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+and doesn't know if it will be faster in bash or python.
+To keep his master branch safe he will do separate branch
```
Comment (pshriwise) [inline _episodes/07-branches.md]: do --> use

Comment (pshriwise) [inline _episodes/07-branches.md]: branch -> branches

File: _episodes/07-branches.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+~~~
+$ ls
+$ git log --oneline
+~~~
+{: .bash}
+
+
+Now we can repeat the process with the `bashdev` branch.
+This time lets create and switch two the `bashdev` branch
```
Comment (pshriwise) [inline _episodes/07-branches.md]: lets -> let's

File: _episodes/07-branches.md (hunk 1 above)
Comment (sstevens2) [inline _episodes/07-branches.md]: :+1:  I was struggling with the objectives.  Thanks!

**Comment by sstevens2** [comment]
Thanks for all the great feedback!  I made suggested changes and updated the PR

**Comment by pshriwise** [comment]
Looks good. Thanks @sstevens2!

**Pull request merged**